skbeam/core/roi.py

```diff
@@ -511,3 +505,138 @@ def extract_label_indices(labels):
     label_mask = labels[labels > 0]
 
     return label_mask, pixel_list
+
+
+def _make_roi(coords, edges, shape):
+    """ Helper function to create ring rois and bar rois
+
+    Parameters
+    ----------
+    coords : array
+        shape is image shape
+    edges : list
+        List of tuples of inner (left or top) and outer (right or bottom)
+        edges of each roi.
+        e.g., edges=[(1, 2), (11, 12), (21, 22)]
+    shape : tuple
+        Shape of the image in which to create the ROIs
+        e.g., shape=(512, 512)
+
+    Returns
+    -------
+    label_array : array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, corresponding to the order they are
+        specified in `edges`.
+        Has shape=`image shape`
+    """
+    label_array = np.digitize(coords, edges, right=False)
+    # Even elements of label_array are in the space between rings.
+    label_array = (np.where(label_array % 2 != 0, label_array, 0) + 1) // 2
+    return label_array.reshape(shape)
+
+
+def bar(edges, shape, horizontal=True, values=None):
+    """Draw bars defined by `edges` from one edge to the other of `image_shape`
+
+    Bars will be horizontal or vertical depending on the value of `horizontal`
+
+    Parameters
+    ----------
+    edges : list
+        List of tuples of inner (left or top) and outer (right or bottom)
+        edges of each bar.
+        e.g., edges=[(1, 2), (11, 12), (21, 22)]
+    shape : tuple
+        Shape of the image in which to create the ROIs
+        e.g., shape=(512, 512)
+    horizontal : bool, optional
+        True: Make horizontal bars
+        False: Make vertical bars
+        Defaults to True
+    values : array, optional
+        image pixels co-ordinates
+
+    Returns
+    -------
+    label_array : array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, corresponding to the order they are
+        specified in `edges`.
+        Has shape=`image shape`
+
+    Note
+    ----
+    The primary use case is in GISAXS.
+    """
+    edges = np.atleast_2d(np.asarray(edges)).ravel()
+    if not 0 == len(edges) % 2:
+        raise ValueError("edges should have an even number of elements, "
```

Comment:
I already added a test for this line to find the `edges to have even number of elements`. line 125 already covered but not line 127 I am adding one now